### ISSUE TITLE
What's the differences between acorn build in rollup and real acorn?

### ISSUE THREAD

**Issue by LongTengDao** [rollup/rollup#3471]
### Feature Use Case

There are many plugins can't use with rollup, just because rollup's built-in acorn edition.

### Feature Proposal

I want to know, what's the necessity to build acorn in rollup? Is it possible to leave it in dependenciess?

Now I use `acornInjectPlugins: [ () => require('acorn').Parser ]` to hack that, but I'm not sure, will it cause rollup bug?

**Comment by lukastaegert** [comment]
It is possible but it was a huge pain and caused its own bunch of issues. The way those broken plugins work is that they require acorn by themselves. Which may give them the correct acorn instance, or it may not. This caused quite a few issues not only for Rollup but also for Webpack (there was one issue with >100 people chiming in that was not really resolved for a very long time). Moreover, it made it impossible for us to support plugins in the browser build. And both installation and startup time of Rollup is now faster.

The "fix" for these plugins is really simple and can be done via pull request by anyone. It is even linked in the [release notes](https://github.com/rollup/rollup/releases/tag/v2.0.0).

Basically it is putting this one-liner in the right place: https://github.com/acornjs/acorn/pull/870#issuecomment-527339830

The decision was to put pressure on the ecosystem so that these plugins are fixed at last and the aforementioned issues are resolved for good.

**Comment by lukastaegert** [comment]
To answer the original question, there is difference. Rollup aims to always bundle the latest version of acorn.

**Comment by lukastaegert** [comment]
Another point to add is that it would currently make Rollup unuseable in a Node 13+ ESM environment, at least as far as I tried.

**Comment by LongTengDao** [comment]
I see. So rollup just bundle the latest version acorn inside to enhance cooperativity, no any other hack modify to acorn api to achieve rollup internal aim which original acorn can't do?

**Comment by lukastaegert** [comment]
Exactly. It is completely unmodified except that it is hoisted into the remaining Rollup bundle.

**Comment by LongTengDao** [comment]
OK, then there will be no problem!
